xscope_fileio/src/xscope_io_device.c

```diff
@@ -195,3 +208,15 @@ void xscope_close_all_files(void){
     if(VERBOSE) printf("Sent close files\n");
     hwtimer_t t = hwtimer_alloc(); hwtimer_delay(t, 5000000); //50ms to allow messages to make it before xgdb quit
 }
+
+void xscope_fclose(xscope_file_t *xscope_file){
+    xscope_fileio_lock_acquire();
+    const unsigned char idx = xscope_file->index + '0';
+    xscope_bytes(XSCOPE_ID_HOST_CLOSE, 1, &idx);
+    if(VERBOSE) {
+        printf("Sent close file id: %d\n", xscope_file->index);
+    }
+    reset_available_file_idx(xscope_file->index);
+    delay_ticks(10); // sanity time to close file
```

Comment:
Be interesting to know where 10 comes from

Reply:
in reality is not needed, I was thinking to give a small time to the next file, but in reallity always that th lock is adquired we wont be able to open a file while one is closing on device.